smbios/stream_windows.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017-2018 DigitalOcean.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package smbios
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"syscall"
+	"unsafe"
+)
+
+const (
+	firmwareTableProviderSigRSMB uint32 = 0x52534d42 // 'RSMB' in ASCII
+)
+
+var (
+	libKernel32 = syscall.NewLazyDLL("kernel32.dll")
+
+	procGetSystemFirmwareTable = libKernel32.NewProc("GetSystemFirmwareTable")
+)
+
+// WindowsEntryPoint contains SMBIOS Table entry point data returned from
+// GetSystemFirmwareTable. As raw access to the underlying memory is not given,
+// the full bredth of information is not available.
+type WindowsEntryPoint struct {
+	Size         uint32
+	MajorVersion byte
+	MinorVersion byte
+	Revision     byte
+}
+
+// Table implements EntryPoint. The returned address will always be 0, as it
+// is not returned by GetSystemFirmwareTable.
+func (e *WindowsEntryPoint) Table() (address, size int) {
+	return 0, int(e.Size)
+}
+
+// Version implements EntryPoint.
+func (e *WindowsEntryPoint) Version() (major, minor, revision int) {
+	return int(e.MajorVersion), int(e.MinorVersion), int(e.Revision)
+}
+
+func stream() (io.ReadCloser, EntryPoint, error) {
+	// Call first with empty buffer to get size
+	r1, _, err := procGetSystemFirmwareTable.Call(
+		uintptr(firmwareTableProviderSigRSMB),
+		uintptr(0),
```

Comment:
I believe you can omit these three conversions here and just pass constant `0` and the compiler will infer the right type.